src/tiled/newmapdialog.ui

```diff
@@ -234,6 +234,13 @@
       <item row="2" column="1">
        <widget class="QComboBox" name="renderOrder"/>
       </item>
+      <item row="3" column="1">
+       <widget class="QCheckBox" name="mapInfinite">
+        <property name="text">
+         <string>Automatically Resize Map</string>
```

Comment:
Please see if this checkbox fits in the "Map Size" box, and disable the width/height inputs and hide the pixel size display for infinite maps (I think the map size should just be 0,0).